lib/fastlane/actions/changelog_from_git_commits.rb

```diff
@@ -0,0 +1,79 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      FL_CHANGELOG = :FL_CHANGELOG
+    end
+
+    class ChangelogFromGitCommitsAction < Action
+      def self.run(params)
+        if params[:between]
+          from, to = params[:between]
+        else
+          from = Actions.last_git_tag_name(params[:match_lightweight_tag])
+          Helper.log.debug "Found the last Git tag: #{from}"
+          to = 'HEAD'
+        end
+
+        Helper.log.info "Collecting Git commits between #{from} and #{to}".green
+
+        changelog = Actions.git_log_between(params[:pretty], from, to)
+        Actions.lane_context[SharedValues::FL_CHANGELOG] = changelog
+
+        changelog
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Collect git commit messages into a changelog"
+      end
+
+      def self.details
+        "By default, messages will be collected back to the last tag, but the range can be controlled"
+      end
+
+      def self.output
+        ['FL_CHANGELOG', 'The changelog String generated from the collected Git commit messages']
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :between,
+                                       env_name: 'FL_GIT_COLLECT_MESSAGES_BETWEEN',
+                                       description: 'Array containing two Git revision values between which to collect messages',
+                                       optional: true,
+                                       is_string: false,
+                                       verify_block: proc do |value|
+                                         raise "'between' must be an array of size 2".red unless value and value.size == 2
```

Comment:
It is recommended to not use the `and` keyword after an inline `if` or `unless`
```ruby
unless value and value.size == 2
```
instead it's safer to use 
```ruby
unless value && value.size == 2
```
There is a different in the priority of those 2 keywords.

To make it more Ruby like you could even use

```ruby
It is recommended to not use the `and` keyword after an inline `if` or `unless`
```ruby
unless value and value.size == 2
```
instead it's safer to use 
```ruby
unless value && value.size == 2
```ruby
unless (value || []).count == 2
```
also you could add a check for the type before that line to fail if the user passes e.g. a number

```ruby
raise ":between must be of type array" unless value.kind_of?(Array)
```

:+1: